networkx/algorithms/tree/mst.py

```diff
@@ -218,18 +306,12 @@ def maximum_spanning_edges(G, algorithm='kruskal', weight='weight', data=True):
 
 @not_implemented_for('directed')
 def _optimum_spanning_tree(G, algorithm, minimum, weight='weight'):
-    try:
-        algo = ALGORITHMS[algorithm]
-    except KeyError:
-        msg = '{} is not a valid choice for an algorithm.'.format(algorithm)
-        raise ValueError(msg)
-
-    edges = algo(G, minimum=minimum, weight=weight, data=True)
+    edges = _spanning_edges(G, minimum, algorithm=algorithm, weight=weight)
     T = nx.Graph(edges)
 
     # Add isolated nodes
     if len(T) != len(G):
-        T.add_nodes_from([n for n, d in G.degree() if d == 0])
+        T.add_nodes_from(nx.isolates(G))
```

Comment:
I always get an error when I try to use an existing function that doesn't exist.